openverse_catalog/dags/providers/provider_api_scripts/smk.py

```diff
@@ -1,223 +1,180 @@
+        except (IndexError, KeyError):
+            return
+
+    @staticmethod
+    def _get_alternative_images(item: dict) -> list:
+        images = []
+
+        # Legacy images do not have an iiif_id; fall back to the ID from the
+        # collection DB.
+        iiif_id = item.get("image_iiif_id")
+        image_id = iiif_id or item.get("id")
+
+        if image_id is not None:
+            if iiif_id is None:
+                # Legacy images do not have IIIF links.
+                image_url = item.get("image_native")
+            else:
+                image_url = SmkDataIngester._get_image_url(iiif_id)
+
+            height = item.get("image_height")
+            width = item.get("image_width")
+            filesize = item.get("image_size") or item.get("size")
+            images.append(
+                {
+                    "id": image_id,
+                    "image_url": image_url,
+                    "height": height,
+                    "width": width,
+                    "filesize": filesize,
+                }
+            )
 
-HEADERS = {"Accept": "application/json"}
+        alternative_images = item.get("alternative_images")
+        if type(alternative_images) == list:
+            for alt_img in alternative_images:
+                if type(alt_img) == dict:
+                    iiif_id = alt_img.get("iiif_id")
+                    if iiif_id is None:
+                        # The API for alternative images does not include the
+                        # 'id', so we must skip if `iiif_id` is not present.
+                        continue
+                    image_url = SmkDataIngester._get_image_url(iiif_id)
+                    height = alt_img.get("height")
+                    width = alt_img.get("width")
+                    filesize = alt_img.get("image_size") or alt_img.get("size")
+
+                    images.append(
+                        {
+                            "id": iiif_id,
+                            "image_url": image_url,
+                            "height": height,
+                            "width": width,
+                            "filesize": filesize,
+                        }
+                    )
+        return images
+
+    @staticmethod
+    def _get_metadata(item: dict) -> dict:
+        meta_data = {}
+        if created_date := item.get("created"):
+            meta_data["created_date"] = created_date
+        collection = item.get("collection")
+        if type(collection) == list:
+            meta_data["collection"] = ",".join(collection)
+        techniques = item.get("techniques")
+        if type(techniques) == list:
+            meta_data["techniques"] = ",".join(techniques)
+        colors = item.get("colors")
+        if type(colors) == list:
+            meta_data["colors"] = ",".join(colors)
+        return meta_data
+
+    def get_record_data(self, data: dict) -> dict | list[dict] | None:
+        license_info = get_license_info(license_url=data.get("rights"))
```

Comment:
If `license_info` is `None` (or similar), could/should we potentially return early?

Reply:
We could! Done in [`ae03ad3`](https://github.com/WordPress/openverse-catalog/pull/742/commits/ae03ad3e3684dd5e29531ff10e2514311b0ea5e3).